api/v3/Case/Getdetails.php

```diff
@@ -155,23 +155,19 @@ function civicrm_api3_case_getdetails(array $params) {
           $categories[$grouping][] = $allTypes[] = $val['value'];
         }
       }
-      $activities = civicrm_api3('Activity', 'get', [
-        'return' => [
-          'activity_type_id', 'subject', 'activity_date_time', 'status_id',
-          'case_id', 'target_contact_name', 'assignee_contact_name',
-          'is_overdue', 'is_star', 'file_id', 'tag_id.name',
-          'tag_id.description', 'tag_id.color',
-        ],
+
+      // Get last activity.
+      $lastActivity = _civicrm_api3_case_get_activities($ids, 1, [
+        'check_permissions' => !empty($params['check_permissions']),
```

Comment:
This line is common, and can be moved inside `_civicrm_api3_case_get_activities` function.

Reply:
Yes this line is common, but don't confuse `$params` here (`civicrm_api3_case_getdetails()` function) with `$params` in `_civicrm_api3_case_get_activities` function.
So any way we need to pass this value to `_civicrm_api3_case_get_activities()` somehow, and I would prefer to do it like this.